src/Http/Http/src/Internal/RequestCookieCollection.cs

```diff
@@ -116,17 +127,20 @@ public bool ContainsKey(string key)
             {
                 return false;
             }
-            return Store.ContainsKey(key);
+            return Store.ContainsKey(key)
+                || !_enableCookieNameDecoding && Store.ContainsKey(Uri.EscapeDataString(key));
         }
 
         public bool TryGetValue(string key, out string value)
         {
-            if (Store == null)
+            if (Store == null || key == null)
```

Comment:
what did Store.TryGetValue with a null key do before?

Reply:
ArgumentNullException. I guess EscapeDataString would do the same I could remove that check.